packages/backend/src/open_payments/quote/service.ts

```diff
@@ -195,55 +192,87 @@ export async function resolveReceiver(
   return receiver
 }
 
-interface CalculateQuoteAmountsAfterFeesResult {
+interface CalculateQuoteAmountsWithFeesResult {
   receiveAmountValue: bigint
   debitAmountValue: bigint
 }
 
-function calculateQuoteAmountsAfterFees(
+/**
+ * Calculate fixed-send quote amounts: debitAmount is locked,
+ * subtract fees (considering the exchange rate) from the receiveAmount.
+ */
+function calculateFixedSendQuoteAmounts(
   deps: ServiceDependencies,
   quote: Quote,
-  maxReceiveAmountValue?: bigint
-): CalculateQuoteAmountsAfterFeesResult {
-  let debitAmountValue = quote.debitAmount.value
-  let receiveAmountValue = quote.receiveAmount.value
-
-  if (!maxReceiveAmountValue) {
-    // FixedDelivery
-    const fees = quote.fee?.calculate(debitAmountValue) ?? 0n
-    debitAmountValue = BigInt(debitAmountValue) + fees
-    if (
-      debitAmountValue < quote.debitAmount.value ||
-      debitAmountValue <= BigInt(0)
-    ) {
-      throw QuoteError.InvalidAmount
-    }
-  } else {
-    // FixedSend
-    const fees = quote.fee?.calculate(receiveAmountValue) ?? 0n
-    const estimatedExchangeRate =
-      quote.estimatedExchangeRate || quote.lowEstimatedExchangeRate.valueOf()
-
-    const exchangeAdjustedFees = BigInt(
-      Math.ceil(Number(fees) * estimatedExchangeRate)
+  maxReceiveAmountValue: bigint
+): CalculateQuoteAmountsWithFeesResult {
+  const fees = quote.fee?.calculate(quote.receiveAmount.value) ?? BigInt(0)
+
+  const estimatedExchangeRate =
+    quote.estimatedExchangeRate || quote.lowEstimatedExchangeRate.valueOf()
+
+  const exchangeAdjustedFees = BigInt(
+    Math.ceil(Number(fees) * estimatedExchangeRate)
+  )
+  const receiveAmountValue =
+    BigInt(quote.receiveAmount.value) - exchangeAdjustedFees
+
+  if (receiveAmountValue <= BigInt(0)) {
+    deps.logger.info(
+      { fees, exchangeAdjustedFees, estimatedExchangeRate, receiveAmountValue },
+      'Negative receive amount when calculating quote amount'
     )
-    receiveAmountValue -= exchangeAdjustedFees
+    throw QuoteError.NegativeReceiveAmount
+  }
 
-    if (receiveAmountValue <= exchangeAdjustedFees) {
-      throw QuoteError.NegativeReceiveAmount
-    }
```

Comment:
Behaviour change:
We used to throw if the receiveAmount was less than the fees: e.g. if receiveAmountValue = 100, and exchangeAdjustedFees were 60, we would throw, even though receiveAmountValue of 40 seems like its ok (just more than 50% of the transfer goes to fees, which wouldn't be nice for the user, but not necessarily a problem)

After a [discussion](https://interledgerfoundation.slack.com/archives/C06KZCKEEKW/p1716562815852079),